jdi-light/src/main/java/com/epam/jdi/light/driver/WebDriverFactory.java

```diff
@@ -11,91 +12,126 @@
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.apache.commons.lang3.StringUtils.isNotBlank;
 
 /**
  * Created by Roman Iovlev on 26.09.2019
  * Email: roman.iovlev.jdi@gmail.com; Skype: roman.iovlev
  */
 public class WebDriverFactory {
-    private WebDriverFactory() { }
+    private static WebDriverFactory driverFactory;
+    private final MapArray<String, JFunc<WebDriver>> DRIVERS;
+    private final boolean SINGLE_THREAD;
+    private MapArray<String, WebDriver> RUN_DRIVERS;
+    private final Safe<MapArray<String, WebDriver>> THREAD_RUN_DRIVERS;
 
-    public static MapArray<String, JFunc<WebDriver>> DRIVERS
-        = new MapArray<>(DEFAULT_DRIVER, () -> initDriver(CHROME));
-    public static boolean SINGLE_THREAD = false;
-    private static MapArray<String, WebDriver> RUN_DRIVERS = new MapArray<>();
-    private static Safe<MapArray<String, WebDriver>> THREAD_RUN_DRIVERS
-        = new Safe<>(MapArray::new);
+    private long INIT_THREAD_ID;
+    private boolean SWITCH_THREAD;
+    private WebDriver INIT_DRIVER;
 
-    public static boolean noRunDrivers() {
+    private WebSettings webSettings = WebSettings.getWebSettings();
+
+
+    private WebDriverFactory() {
+        RUN_DRIVERS = new MapArray<>();
+        THREAD_RUN_DRIVERS = new Safe<>(MapArray::new);
+        DRIVERS = new MapArray<>(DEFAULT_DRIVER, () -> initDriver(getByName(getJDISettings().DRIVER.name)));
+        SINGLE_THREAD = false;
+        INIT_THREAD_ID = -1;
+        SWITCH_THREAD = false;
+    }
+
+    public static WebDriverFactory getWebDriverFactory() {
+        if (driverFactory == null) {
+            driverFactory = new WebDriverFactory();
+        }
+        return driverFactory;
+    }
+
+    public MapArray<String, JFunc<WebDriver>> getDrivers() {
+        return DRIVERS;
+    }
+
+    public long getInitThreadId() {
+        return INIT_THREAD_ID;
+    }
+
+    public void setInitThreadId(long Id) {
+        INIT_THREAD_ID = Id;
+    }
+
+    public boolean noRunDrivers() {
         return !getRunDrivers().any();
     }
-    private static MapArray<String, WebDriver> getRunDrivers() {
-        logger.debug("SINGLE_THREAD=" + SINGLE_THREAD);
+
+    private MapArray<String, WebDriver> getRunDrivers() {
+        webSettings.logger.debug("SINGLE_THREAD=" + SINGLE_THREAD);
         return SINGLE_THREAD
-            ? RUN_DRIVERS
-            : THREAD_RUN_DRIVERS.get();
+                ? RUN_DRIVERS
+                : THREAD_RUN_DRIVERS.get();
     }
-    private static void setRunDrivers(MapArray<String, WebDriver> map) {
+
+    private void setRunDrivers(MapArray<String, WebDriver> map) {
         if (SINGLE_THREAD) {
             RUN_DRIVERS = map;
         } else {
             THREAD_RUN_DRIVERS.set(map);
         }
     }
-    public static WebDriver getDriverByName(String driverName) {
-        logger.debug("getDriverByName(%s)", driverName);
+
+    public WebDriver getDriverByName(String driverName) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The method 'getDriverByName(String)' has an NPath complexity of 208](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5482899)